src/distributed-transaction/percolator.md

```diff
@@ -0,0 +1,150 @@
+# Percolator
+
+TiKV supports distributed transaction, which is inspired by Google's paper about *Percolator*. We will introduce it and how we make use of it in TiKV in this section.
+
+## What is Percolator?
+
+*Percolator* is a system used for incremental processing on very large data set, built by Google. Support for distributed transactions is only a part of it. We will introduce Percolator briefly here. You can view the full paper [here](https://ai.google/research/pubs/pub36726#), and if you are very familiar to this paper, you can skip this section and go to read [Percolator in TiKV](#Percolator-in-TiKV)
+
+Percolator is built based on Google's BigTable. BigTable is a distributed storage system that supports single-row transactions, based on which Percolator implements distributed transactions in ACID snapshot-isolation semantics. A column `c` of Percolator is actually divided into following internal columns of BigTable:
+
+* `c:lock`
+* `c:write`
+* `c:data`
+* `c:notify`
+* `c:ack_O`
+
+Percolator also relies on a service named *timestamp oracle*. The timestamp oracle can produces timestamps in strictly increasing order. All read and write operations need to apply for timestamps from the timestamp oracle, and a timestamp came from timestamp oracle will be regarded as the logical time when the read/write operation happens.
+
+Percolator is multi-version, and a data item's version is represented by the timestamp that the transaction was committed.
+
+For example, in such a state:
+
+<table>
+  <tr>
+    <th>key</th><th>v:data</th><th>v:lock</th><th>v:write</th>
+  </tr>
+  <tr>
+    <td rowspan="3">k1</td>
+    <td>14: "value2"</td><td>14: primary</td><td></td>
+  </tr>
+  <tr>
+    <td></td><td></td><td>12: data@10</td>
+  </tr>
+  <tr>
+    <td>10: "value1"</td><td></td><td></td>
+  </tr>
+</table>
+
+It means that for the key `k1`, a value `"value1"` was committed at timestamp `12`. Then there is an uncommitted version whose value is `"value2"`, and it's uncommitted because there's a lock. You will understand why it looks like this after understanding how transactions works.
```

Comment:
```suggestion
It means that for the key `k1`, a value `"value1"` was committed at timestamp `12`. Then there is an uncommitted version whose value is `"value2"`, and it's uncommitted because there's a lock. You will understand why it looks like this after understanding how transactions work.
```